src/pcp/types.rs

```diff
@@ -217,6 +217,133 @@ impl<F: FieldElement> Value for Sum<F> {
     }
 }
 
+/// The histogram type. Each measurement is a non-negative integer and the aggregate is a histogram
+/// approximating the distribution of the measurements.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Histogram<F> {
+    data: Vec<F>,
```

Comment:
IIUC, we represent a single contribution to a histogram as a one-hot bit vector whose dimension is the number of buckets, where we use a field for each bit. Could we compress this into the index of the bucket that the value falls into, which would require a single field? That said, the range checker polynomial would have to be of degree equal to the number of buckets, but maybe that's tenable for most histograms which will have a small number of buckets.

Reply:
If this is possible at all, we could/should save it for later. Maybe do something clever where we use either the bucket index only for histograms with few enough buckets, and the bit vector otherwise.

Reply:
Remember: the size of the proof is proportional to the degree of this polynomial, so something like this wouldn't yield any savings, at least asymptotically.

That said, we can definitely do better.
* By replacing the input share with a DPF key share, we can get the input size down from O(N) to O(lon N) where N is the number of buckets.
* For the proof we can do one of the following:
  * We can replace the Prio-style input validation step with the secure sketching protocol from the heavy hitters protocol. That would get the size of the proof down from O(N) to O(log N). The downside is that it requires two rounds instead of one.
  * Using something like #37 we can get the proof size down to O(N^0.5).

However, as you pointed out, N (the number of buckets) is going to be relatively small for most applications (on the order of 10 to 100 is what I'd imagine.) I definitely want to implement proof-size optimizations for the vectorized variants of things (see #94), but since these come at the cost of a certain amount of computational overhead, perhaps it makes sense to hold off here.

Reply:
Oh, right, I forgot about the proof size! OK, let's keep this as-is.